packages/plugins/ens/src/index.js

```diff
@@ -90,17 +90,32 @@ class ENS {
     this.embark.registerActionForEvent("pipeline:generateAll:before", this.addArtifactFile.bind(this));
   }
 
-  getEnsConfig() {
-    return {
-      env: this.env,
-      registration: this.config.namesystemConfig.register,
-      registryAbi: this.ensConfig.ENSRegistry.abiDefinition,
-      registryAddress: this.ensConfig.ENSRegistry.deployedAddress,
-      registrarAbi: this.ensConfig.FIFSRegistrar.abiDefinition,
-      registrarAddress: this.ensConfig.FIFSRegistrar.deployedAddress,
-      resolverAbi: this.ensConfig.Resolver.abiDefinition,
-      resolverAddress: this.ensConfig.Resolver.deployedAddress
-    };
+  getEnsConfig(done) {
+    async.map(
+      this.config.namesystemConfig.dappConnection || this.config.contractsConfig.dappConnection,
+      (conn, next) => {
+      if (conn === '$EMBARK') {
+        return this.events.request('proxy:endpoint:get', (e, val) => {
+          next(e, val);
+        });
+      }
+      next(null, conn);
+    }, (err, connections) => {
```

Comment:
We have to replace `$EMBARK` with the actual proxy endpoint **after** the proxy is actually booted up and **before** we register the `__embarkjsENS.setProvider()` in the console because it'll try to connect to whatever endpoint is available.